zanata-war/src/main/java/org/zanata/action/VersionGroupJoinAction.java

```diff
@@ -149,14 +152,45 @@ public String send() {
         }
     }
 
-    @AllArgsConstructor
-    public final class SelectableProject {
+    /**
+     * This is to reset data when user closes dialog or after sending email.
+     * See version-group/request_join_modal.xhtml#cancelJoinGroupEmail
+     */
+    public void resetData() {
+        projectSlug = "";
+        projectVersions.clear();
+        setQuery("");
+    }
 
-        @Getter
-        private HProjectIteration projectIteration;
+    // clear all the project versions in the selected project
+    private void clearProjectVersions() {
+        if(!projectVersions.isEmpty()) {
+            projectVersions.clear();
+        }
+    }
+
+    @Override
+    public List<HProject> suggest() {
+        clearProjectVersions();
+        return projectDAO.getProjectsForMaintainer(
+                authenticatedAccount.getPerson(), getQuery(), 0,
+                Integer.MAX_VALUE);
```

Comment:
It looks like you just changed from `projectVersions.clear();` to `clearProjectVersions();`. I think it was already clear what that does. The question is why?

> It looks like this method is supposed to just return a list of projects, but it is also clearing the versions. I think some code needs to be restructured so that this is not confusing, or maybe add a comment saying why it needs to clear the versions.